src/logics/ArithLogic.h

```diff
@@ -266,7 +266,7 @@ class ArithLogic: public Logic
 
     bool isNumVar(SymRef sr) const { return isVar(sr) and (yieldsSortInt(sr) or yieldsSortReal(sr)); }
     bool isNumVar(PTRef tr) const { return isNumVar(getPterm(tr).symb()); }
-    bool isNumVarLike(SymRef sr) const { return isNumVar(sr) or isIte(sr) or isIntDiv(sr) or isMod(sr) or (hasUFs() and isUF(sr)); }
+    bool isNumVarLike(SymRef sr) const { return yieldsSortNum(sr) and not isPlus(sr) and not isTimes(sr) and not isNumConst(sr); }
```

Comment:
Why there are so much differences between conditions?

Reply:
The previous version enumerated what should be included. The new version enumerates what should be excluded (included is anything that `yieldsSortNum` as the starting point).

This change means to include array select terms as `numVarLike`.